### ISSUE TITLE
[top, aes] Disable `SecAllowForcingMasks` parameter

### ISSUE THREAD

**Issue by vogelpi** [lowRISC/opentitan#15674]
With #15639 we have enabled this parameter for ES (the reasoning is documented in #14240). For production, we will need to disable it again.

**Comment by vogelpi** [comment]
I believe we still want `SecAllowForcingMasks` enabled for M2.5 but we should be absolutely sure about this. I would like to re-discuss this in an upcoming Security meeting.

For reference, [this comment outlines how difficult it is to disable the masking in the field](https://github.com/lowRISC/opentitan/issues/14240#issuecomment-1297706702).

**Comment by johannheyszl** [comment]
@vogelpi and I discussed and are sure we need this for M2.5

**Comment by msfschaffner** [comment]
@vogelpi and @johannheyszl can you determine what the plan should be for this in PROD?

**Comment by vogelpi** [comment]
Yes, @johannheyszl and I believe we should keep this in for production due to the big RTL change around the PRNG. The whole feature can be effectively gated with ROM_ext and even if that was bypassed, it's fairly hard to make use of the feature, see https://github.com/lowRISC/opentitan/issues/14240#issuecomment-1293222494.

**Comment by vogelpi** [comment]
For Earlgrey-PROD, the decision is taken to keep this and I am removing the corresponding labels. We should keep the issue around for other designs such as Darjeeling.
